src/encoder.rs

```diff
@@ -61,64 +66,97 @@ where
     }
 
     pub fn with_chunks_size(output: W, chunks: usize) -> Encoder<W> {
-        Encoder {
+        let chunks_size = if chunks > MAX_CHUNK_SIZE {
+            MAX_CHUNK_SIZE
+        } else {
+            chunks
+        };
```

Comment:
are these lines the same as:

```suggestion
        let chunks_size = chunks.min(MAX_CHUNK_SIZE);
```

if so, thoughts on using it?

Reply:
Ooh, I didn't know about this method! Thanks for mentioning it. Do you know where it's defined? I couldn't find documentation for it on usize or in the `std::cmp` package.

Reply:
it comes from the [`Ord`](https://doc.rust-lang.org/nightly/std/cmp/trait.Ord.html#method.min) trait which is implemented for `usize`